src/toggle/__tests__/toggle.test.tsx

```diff
@@ -100,3 +103,55 @@ test('check a11y', async () => {
   );
   await expect(wrapper.getElement()).toValidateA11y();
 });
+
+test('Should set aria-describedby and aria-labelledby from Formfield', () => {
+  const { container } = render(
+    <FormField description="This is a formfield description." label="Form field label">
+      <Toggle checked={false} description="This is description">
+        Toggle label
+      </Toggle>
+    </FormField>
+  );
+  const formFieldWrapper = createWrapper(container).findFormField();
+  const toggleWrapper = createWrapper(container).findToggle()!;
+  const toggleAriaDescribedby = toggleWrapper.findNativeInput().getElement().getAttribute('aria-describedby');
+  const toggleAriaLabelledby = toggleWrapper.findNativeInput().getElement().getAttribute('aria-labelledby');
+
+  expect(toggleAriaDescribedby).toBe(
+    joinStrings(
+      formFieldWrapper?.findDescription()?.getElement().id,
+      container?.querySelector(`.${abstractSwitchStyles.description}`)?.id
+    )
+  );
+  expect(toggleAriaLabelledby).toBe(
+    joinStrings(
+      container?.querySelector(`.${abstractSwitchStyles.label}`)?.id,
+      formFieldWrapper?.findLabel()?.getElement().id
+    )
+  );
+});
+
+test('Should set aria-describedby and aria-labelledby from ariaLabelledby and ariaDescribedby', () => {
+  const { container } = render(
+    <FormField description="This is a description." label="Form field label">
+      <div id="label-id">it is label</div>
+      <div id="description-id">it is label</div>
+      <Toggle
+        checked={false}
+        description="This is description"
+        ariaLabelledby="label-id"
+        ariaDescribedby="description-id"
+      />
+    </FormField>
+  );
+  const toggleWrapper = createWrapper(container).findToggle()!;
+  const toggleAriaDescribedby = toggleWrapper.findNativeInput().getElement().getAttribute('aria-describedby');
+  const toggleAriaLabelledby = toggleWrapper.findNativeInput().getElement().getAttribute('aria-labelledby');
+
+  expect(toggleAriaDescribedby).toBe(
+    joinStrings('description-id', container?.querySelector(`.${abstractSwitchStyles.description}`)?.id)
+  );
+  expect(toggleAriaLabelledby).toBe(
+    joinStrings(container?.querySelector(`.${abstractSwitchStyles.label}`)?.id, 'label-id')
```

Comment:
The implementation uses `joinStrings` already, so the test shouldn't, or it becomes `expect(add(2, 2)).toEqual(add(2, 2))` which is pointless.

Is there any reason not to put the full computed string here?

Reply:
The point of the test is to check if toggle input has aria-describedby identified with formfield description id **AND** its own description id. So I think it is testing `expect(add(attr1, attr2)).toEqual(add(attr3, atrr4))`. joinString is not the point.

Reply:
Having a static string in the right hand side is still preferred. If you cannot use a static string because there is a generated value, you can use `toMatch` ([example](https://github.com/cloudscape-design/components/blob/7433543fcd2796d6398b67867d25b637c1a2b6de/src/tabs/__tests__/tabs.test.tsx#L137))

```js
expect(toggleAriaLabelledby).toMatch(/^awsui-toggle-\d+ label-id/)
```